exercises/beer-song/canonical-data.json

```diff
@@ -0,0 +1,47 @@
+{
+  "verse": {
+    "description": "return 1 verse",
+    "cases": [
+      {
+        "description": "first verse",
+        "number": 99,
+        "expected": "99 bottles of beer on the wall, 99 bottles of beer.\nTake one down and pass it around, 98 bottles of beer on the wall.\n"
+      },
+      {
+        "description": "middle verse",
+        "number": 44,
+        "expected": "44 bottles of beer on the wall, 44 bottles of beer.\nTake one down and pass it around, 43 bottles of beer on the wall.\n"
+      },
+      {
+        "description": "third to last verse",
+        "number": 2,
+        "expected": "2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n"
+      },
+      {
+        "description": "penultimate verse",
+        "number": 1,
+        "expected": "1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n"
+      },
+      {
+        "description": "last verse",
+        "number": 0,
+        "expected": "No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n"
+      }
+    ]
+  },
+  "lyrics": {
+    "description": "return multiple verses",
+    "cases": [
+      {
+        "description": "last 4 verses",
+        "beginning": 3,
+        "end": 0,
+        "expected": "3 bottles of beer on the wall, 3 bottles of beer.\nTake one down and pass it around, 2 bottles of beer on the wall.\n\n2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n\n1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n\nNo more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n"
+      },
+      {
+        "description": "all verses",
```

Comment:
Sigh, "calling conventions" is a reserved word for how arguments get passed in registers / memory, at the computer architecture level. I'm editing my message and changing "calling conventions" to "function overloads / variadic functions"